### ISSUE TITLE
Checkpoint does not load

### ISSUE THREAD

**Issue by rmsmith88** [mozilla/DeepSpeech#710]
I have tried to restart with only a different dataset and same checkpoint directory and it doesn't load the old checkpoint. Is there anything i am missing?

**Comment by kdavis-mozilla** [comment]
What command line parameters did you pass to the system?

**Comment by rmsmith88** [comment]
```
/share/apps/libc6_2.17/lib/x86_64-linux-gnu/ld-2.17.so $(command -v /share/apps/python-3.6.0-shared/bin/python3) /cluster/project2/rob/DeepS
peech/DeepSpeech.py --train_files /cluster/project2/rob/DeepSpeech/data/ted/ted-train.csv --dev_files /cluster/project2/rob/DeepS
peech/data/ted/ted-dev.csv --test_files /cluster/project2/rob/DeepSpeech/data/ted/ted-test.csv --train_batch_sze 12 --dev_batch_size 12 --te
st_batch_size 12 --learning_rate 0.0001 --epoch 15 --display_step 5 --validation_step 5 --dropout_rate 0.30 --default_stddev 0.046875 --checkpoint_dir
/home/rsmith/.local/share/deepspeech/tf-libivox-full --checkpoint_secs 1200 --summary_secs 900 --log_level 0
```

**Comment by kdavis-mozilla** [comment]
What is in the checkpoint directory? In other words what's the output of
```bash
rsmith@localhost:~$ ls -lrt /home/rsmith/.local/share/deepspeech/tf-libivox-full
```

**Comment by rmsmith88** [comment]
```
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul  9 12:21 model.ckpt-46753.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul  9 12:21 model.ckpt-46753.index
-rw-r--r-- 1 rsmith cscluster    1701916 Jul  9 12:21 model.ckpt-46753.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul  9 12:31 model.ckpt-46785.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul  9 12:31 model.ckpt-46785.index
-rw-r--r-- 1 rsmith cscluster    1701916 Jul  9 12:31 model.ckpt-46785.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul  9 12:42 model.ckpt-46816.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul  9 12:42 model.ckpt-46816.index
-rw-r--r-- 1 rsmith cscluster    1701916 Jul  9 12:42 model.ckpt-46816.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul  9 12:52 model.ckpt-46846.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul  9 12:52 model.ckpt-46846.index
-rw-r--r-- 1 rsmith cscluster    1701916 Jul  9 12:52 model.ckpt-46846.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul  9 13:02 model.ckpt-46872.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul  9 13:02 model.ckpt-46872.index
-rw-r--r-- 1 rsmith cscluster    1701916 Jul  9 13:02 model.ckpt-46872.meta
-rw-r--r-- 1 rsmith cscluster    6450889 Jul  9 13:02 events.out.tfevents.1499263622.mitchell.local
-rw-r--r-- 1 rsmith cscluster    6514497 Jul 10 16:00 events.out.tfevents.1499698727.mitchell.local
-rw-r--r-- 1 rsmith cscluster    6514497 Jul 10 16:09 events.out.tfevents.1499699350.mitchell.local
-rw-r--r-- 1 rsmith cscluster    6554554 Jul 10 18:02 events.out.tfevents.1499705823.mitchell.local
-rw-r--r-- 1 rsmith cscluster    6554650 Jul 10 18:42 events.out.tfevents.1499707485.mitchell.local
-rw-r--r-- 1 rsmith cscluster        285 Jul 12 01:08 checkpoint.bak
-rw-r--r-- 1 rsmith cscluster    6514497 Jul 12 01:22 events.out.tfevents.1499818927.mitchell.local
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 08:08 model.ckpt-14557.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 08:08 model.ckpt-14557.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 08:08 model.ckpt-14557.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 08:28 model.ckpt-14992.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 08:28 model.ckpt-14992.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 08:28 model.ckpt-14992.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 08:48 model.ckpt-15420.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 08:48 model.ckpt-15420.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 08:48 model.ckpt-15420.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 09:08 model.ckpt-15842.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 09:08 model.ckpt-15842.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 09:08 model.ckpt-15842.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 09:28 model.ckpt-16259.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 09:28 model.ckpt-16259.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 09:28 model.ckpt-16259.meta
-rw-r--r-- 1 rsmith cscluster    6556882 Jul 12 09:29 events.out.tfevents.1499819281.mitchell.local
-rw-r--r-- 1 rsmith cscluster    6514497 Jul 12 09:47 events.out.tfevents.1499849202.mitchell.local
-rw-r--r-- 1 rsmith cscluster    3478511 Jul 12 09:50 graph.pbtxt
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 10:11 model.ckpt-1915.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 10:11 model.ckpt-1915.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 10:11 model.ckpt-1915.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 10:31 model.ckpt-3246.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 10:31 model.ckpt-3246.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 10:31 model.ckpt-3246.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 10:51 model.ckpt-4370.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 10:51 model.ckpt-4370.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 10:51 model.ckpt-4370.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 11:11 model.ckpt-5348.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 11:11 model.ckpt-5348.index
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 11:11 model.ckpt-5348.meta
-rw-r--r-- 1 rsmith cscluster 1472790888 Jul 12 11:31 model.ckpt-6225.data-00000-of-00001
-rw-r--r-- 1 rsmith cscluster       1581 Jul 12 11:31 model.ckpt-6225.index
-rw-r--r-- 1 rsmith cscluster        589 Jul 12 11:31 checkpoint
-rw-r--r-- 1 rsmith cscluster    1743813 Jul 12 11:31 model.ckpt-6225.meta
```

**Comment by rmsmith88** [comment]
I also tried to copy the checkpoint file, the latest model index, meta and data files to my local machine and test with some audio I recorded myself and it failed to load the checkpoint. It just starts from 0.

**Comment by tilmankamp** [comment]
@rmsmith88 Did you share the checkpoint directory among different runs of different versions of the code/graph?

**Comment by rmsmith88** [comment]
Yes I shared the directory from different runs, however the computational graph did not change. I (assume) I am using the same graph (based on the same run librivox parameters). The only DeepSpeech parameter change was log_level and summary. 

To prove this is the case I can stop the existing run. Restart it and watch it count from 0.
`1.4G -rw-r--r-- 1 rsmith cscluster 1.4G Jul 12 12:50 model.ckpt-0.data-00000-of-00001
4.0K -rw-r--r-- 1 rsmith cscluster 1.6K Jul 12 12:50 model.ckpt-0.index
1.7M -rw-r--r-- 1 rsmith cscluster 1.7M Jul 12 12:50 model.ckpt-0.meta
1.4G -rw-r--r-- 1 rsmith cscluster 1.4G Jul 12 13:10 model.ckpt-1955.data-00000-of-00001
4.0K -rw-r--r-- 1 rsmith cscluster 1.6K Jul 12 13:10 model.ckpt-1955.index
1.7M -rw-r--r-- 1 rsmith cscluster 1.7M Jul 12 13:10 model.ckpt-1955.meta
`
Next step as discussed in the IRC channel. Start a complete fresh run with a new directory and simple LDC.

**Comment by rmsmith88** [comment]
I restarted with just LDC, created a new checkpoint directory, ran -> it finished. Reran and it did NOT load the previous checkpoint. 

```
/share/apps/libc6_2.17/lib/x86_64-linux-gnu/ld-2.17.so $(command -v /share/apps/python-3.6.0-shared/bin/python3) /cluster/project2/rob/DeepS
peech/DeepSpeech.py --train_files /cluster/project2/rob/DeepSpeech/data/ldc93s1/ldc93s1.csv --dev_files /cluster/project2/rob/DeepSpeech/data/ldc93s1/ldc93s1.csv --test_files/cluster/project2/rob/DeepSpeech/data/ldc93s1/ldc93s1.csv --train_batch_sze 1 --dev_batch_size 1 --test_batch_size 1 --learning_rate 0.0001 --epoch 15 --display_step 1 --validation_step 5 --dropout_rate 0.30 --default_stddev 0.046875 --checkpoint_dir /home/rsmith/.local/share/deepspeech/newtest/ --checkpoint_secs 12 --summary_secs 10 --log_level 0
```
I will retry this locally and not on the server as it could be an issue with the server.

**Comment by rmsmith88** [comment]
Wunderbar, I can confirm that this is now fixed with 862815a:

```
1.7M -rw-r--r-- 1 rsmith cscluster 1.7M Jul  9 12:42 model.ckpt-46816.meta
1.4G -rw-r--r-- 1 rsmith cscluster 1.4G Jul  9 12:52 model.ckpt-46846.data-00000-of-00001
4.0K -rw-r--r-- 1 rsmith cscluster 1.6K Jul  9 12:52 model.ckpt-46846.index
1.7M -rw-r--r-- 1 rsmith cscluster 1.7M Jul  9 12:52 model.ckpt-46846.meta
1.4G -rw-r--r-- 1 rsmith cscluster 1.4G Jul 12 15:19 model.ckpt-46872.data-00000-of-00001
4.0K -rw-r--r-- 1 rsmith cscluster    4 Jul 12 15:19 model.ckpt-46872.data-00000-of-00002
1.4G -rw-r--r-- 1 rsmith cscluster 1.4G Jul 12 15:19 model.ckpt-46872.data-00001-of-00002
4.0K -rw-r--r-- 1 rsmith cscluster 1.6K Jul 12 15:19 model.ckpt-46872.index
1.7M -rw-r--r-- 1 rsmith cscluster 1.7M Jul 12 15:19 model.ckpt-46872.meta
```

Thanks all
